File: pset6/caesar.py
import sys
from cs50 import get_string

# handles command-line orders
if len(sys.argv) != 2 or int(sys.argv[1]) <= 0 :
    print("Usage: python caesar.py k")
    exit(1)

# get the key; the divide operator handles situations when k > 26
key = int(sys.argv[1]) % 26

plaintext = get_string("plaintext: ")

print("ciphertext: ", end="")

for c in plaintext:

# when a character is upper case
    if c.isalpha() and c.isupper():
        if (ord(c) + key) > 90:
            print(chr(ord(c) + key - 26), end="")
        else :
            print(chr(ord(c) + key), end="")

# when a character is lower case
    elif c.isalpha() and c.islower():

        if (ord(c) + key) > 122:
            print(chr(ord(c) + key - 26), end="")
        else:
            print(chr(ord(c) + key), end="")

# everything else
    else:
        print(c, end="")

print("")
exit(0)

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>
//include all necesary libraries

int main(int argc, string argv[])
{
    int j, k;
    string l = argv[1];
    // if there are only 2 arguments in the command, proceed; else go to error message
    if (argc == 2)
    {
        //turn string into integer
        int i = atoi(l);
        string s = get_string("plaintext:  ");
        printf("ciphertext:  ");
        for (j = 0, k = strlen(s); j < k; j++)
        {
            //if the ith letter is upper case
            if (isupper(s[j]))
            {
                int x = (s[j] - 'A' + i) % 26;
                printf("%c", 'A' + x);
            }
            //if the letter is lower case
            else if (islower(s[j]))
            {
                int y = (s[j] - 'a' + i) % 26;
                printf("%c", 'a' + y);
            }
            else
            {
                //everything else
                printf("%c", s[j]);
            }
        }
        printf("\n");
    }
    else
    {
        //error message
        printf("Usage: ./caesar k\n");
        return 1;
    }
}